File: lab2/lab2.py
import matplotlib.pyplot as plt
import math

x1 = float(input("Enter starting co-ordinate x1:"))
y1 = float(input("Enter starting co-ordinate y1:"))
x2 = float(input("Enter ending co-ordinate x2:"))
y2 = float(input("Enter ending co-ordinate y2:"))

dx = x2-x1
dy = y2-y1

xk = x1
yk = y1
m = dy/dx
plt.plot([x1,x2],[y1,y2])
if m<1:
    steps = int(abs(dx))
else:
    steps = int(abs(dy))

xincr = dx/float(steps)
yincr = dy/float(steps)
i = 0
print("iteration\t xk\t\t yk\t (xk+1,yk+1)")
while(i < steps):
    plt.scatter(round(xk),round(yk))
    print(f"{i}\t\t{round(xk,2)}\t\t {round(yk,2)}\t({round(xk)},{round(yk)})")
    i = i + 1
    xk += xincr
    yk += yincr
print("Name : Aditya Pokhrel")
print("Roll no : 3")
plt.show()

File: lab2/lab2.cpp
// WAP to implement DDA algorithm
#include<iostream>
#include<math.h>
using namespace std;

int main()
{
    float x1,y1,x2,y2,dx,dy,m,xi,yi,xk,yk;
    int steps,i;
    cout<<"Enter starting co-ordinates x1 and y1:";
    cin>>x1>>y1;
    cout<<"Enter ending co-ordinates x2 and y2:";
    cin>>x2>>y2;
    xk = x1;
    yk = y1;
    dx = x2 - x1;
    dy = y2 - y1;
    m = dy/dx;
    if(m < 1)
        steps = fabs(dx);
    else if(m > 1)    
        steps = fabs(dy);
    else 
        steps = fabs(dy);
    xi = dx/float(steps);
    yi = dy/float(steps);
    for(i = 0;i < steps;i++)
    {
        cout<<"iteration:"<<i<<"\t"<<"xk:"<<xk<<"\t"<<"yk:"<<yk<<"\t"<<"(xk+1,yk+1):"<<"("<<roundf(xk)<<","<<roundf(yk)<<")"<<endl;
        xk = xk + xi;
        yk = yk + yi;
    }
    return 0;
}
